localstack/services/sqs/provider.py

```diff
@@ -404,6 +413,105 @@ def stop(self):
             self.scheduler = None
 
 
+class MessageMoveTaskManager:
+    """
+    Manages and runs MessageMoveTasks.
+
+    TODO: we should check how AWS really moves messages internally: do they use the API?
+     it's hard to know how AWS really does moving of messages. there are a number of things we could do
+     to understand it better, including creating a DLQ chain and letting move tasks fail to see whether
+     move tasks cause message consuming and create receipt handles. for now, we're doing a middle-layer
+     transactional move, foregoing the API layer but using receipt handles and transactions.
+
+    TODO: restoring move tasks from persistence doesn't work, may be a fringe case though
+    """
+
+    def __init__(self) -> None:
+        self.mutex = threading.RLock()
+        self.move_tasks: dict[str, MessageMoveTask] = dict()
+        self.executor = ThreadPoolExecutor(max_workers=100, thread_name_prefix="sqs-move-message")
+
+    def submit(self, move_task: MessageMoveTask):
+        with self.mutex:
+            move_task.mark_started()
+            source_queue = self._get_queue_by_arn(move_task.source_arn)
+            move_task.approximate_number_of_messages_to_move = (
+                source_queue.approx_number_of_messages
+            )
+            move_task.approximate_number_of_messages_moved = 0
+            self.move_tasks[move_task.task_id] = move_task
+            self.executor.submit(self._run, move_task)
+
+    def cancel(self, move_task: MessageMoveTask):
+        with self.mutex:
+            move_task.status = MessageMoveTaskStatus.CANCELLING
+            move_task.cancel_event.set()
+
+    def close(self):
+        with self.mutex:
+            for move_task in self.move_tasks.values():
+                move_task.cancel_event.set()
+
+            self.executor.shutdown(wait=False)
+
+    def _run(self, move_task: MessageMoveTask):
+        move_task.status = MessageMoveTaskStatus.RUNNING
+
```

Comment:
i was thinking about persistence here, so we can re-start existing move tasks. but since we don't support persistence at all, might as well remove this one here :-) i'll add another try/catch around `start` so it doesn't stay in "RUNNING" if an error occurs before actually starting the task